Hfss importer

SebastianSemper opened the pull request:
This branch should be rebased against dev0.4 if its contents should be present in v0.4. The release v0.3 is very old. Also, we use black as a code formatter, which can on dev0.4 be used via `make black` which makes some of the changes in the commits below obsolete.

ChristophWWagner committed:
Correct typos in docu

ChristophWWagner committed:
Fix encoding problem with german 'Umlaut's

ChristophWWagner committed:
Add the notorious makefile

... featuring useful commands such as
  `make install/uninstall/styleCheck/codeStats/...`

Mitigate line PEP8 continuation errors

Corrected makefile styleCheck configuration

The Travis CI run contained some information about the style guidelines
for this project, which it communicated by failing the extension :)
Therefore the makefile styleCheck configuration was adapted to match
that of the Travis CI suite and errors were corrected.

ChristophWWagner committed:
Add HFSS importer

Unfortunately, verification of the imported data was not completed
fully yet, due to another bug (see #52), which prevents EADF object
initialization for the data contained in the available HFSS exports.
Stay tuned...

ChristophWWagner commented:
good to know, I'll rebase. Please note, that prior to merging, #52 must be resolved first.

ChristophWWagner committed:
Fix codacy issues

ChristophWWagner committed:
Travis CI: Limit deployment to master branch only

ChristophWWagner committed:
Reintroduce exception raises for errors (WIP)

This commit only scopes the `EADF.eadf` submodule. It is suggested to check
out the modifications of this unit, and the corresponding tests in
`test_eadf.py`, prior to extending the modifications to the whole package.

The modifications in this commit tackle issue #56 and propose to apply the
approach detailed here throughout the whole package. Please discuss it in
the thread to issue #56.

Modifications:
--------------
 - Instances of `logging.error` were replaced by raising exceptions
 - A new `TestCase` baseclass was introduced in `test_eadf.py` which offers
   a new method, `assertRaises With Message` that is capable of running some
   code as a contextmanager and check if an exception of proper king was
   raised therein, producing an intended error message. The test fails if
   either no exception was thrown, an exception of the wrong type was thrown,
   or the error message does not match the requirement.
   Checking the exception message is optional.
 - All instances of mocking `logging.error` within `test_eadf.py` was replaced
   by the new assertion mechanisms for exceptions.

ChristophWWagner committed:
Extend EADF.__checkInputData()

This commit tackles issues #53 and #55.

 - Remove code duplicates for checks of input parameter shape and gridding.
 - Introduce handling of SI unit types by extending input parameters to pint
   Quantities.
 - Refactoring the structure of `EADF.__checkInputData(...)` to enable
   modifying the input arrays prior to object initialization. For this,
   passing additional keyworded arguments `EADF.__init__(..., **options)` is
   possible by handing `**options` through to `EADF.__checkInputData()`.

ChristophWWagner committed:
Modify keyworded argument declaration in docs

ChristophWWagner committed:
Correct HFSS function name for polar phase type

ChristophWWagner committed:
Fix problem of dual unit registry

ChristophWWagner committed:
Add preprocessing capabilities to EADF constructor

 - Allows to modify the support values for the angular and frequency
   axes by passing a function to the `**options`:
     `modify_support_CoEle` for the Co-Elevation axis
     `modify_Azi` for the Azimuth axis
     `modify_support_Freq` for the Frequency axis.
   Only the support vector is altered, not the tensor data itself.
   For example, this is useful for adjusting the specification of
   angles in Elevation to Co-Elevation angles.
 - Truncate along angular or frequency axes by providing a (min,max)
   tuple, indicating the cropping range, to the `**options`:
     `truncate_CoEle` for the Co-Elevation axis
     `truncate_Azi` for the Azimuth axis
     `truncate_Freq` for the Frequency axis
 - Sort the tensor data along the angular and frequency axes if
   `sort=True` is given in `**options`. Also sorts the support

ChristophWWagner committed:
Fix missing patch definition in test_eadf.py

ChristophWWagner committed:
Provide HFSS importer usage example

ChristophWWagner committed:
Refactor exceptions in importers and test_importers

ChristophWWagner committed:
Add stylecheck to make test

ChristophWWagner committed:
Refactor exception handling in arrays and test_arrays

ChristophWWagner committed:
Refactor exception raising. Solves #56

SebastianSemper commented:
This branch fails the test for me with:
```
[seb@tn1057 EADF]$ make test
pycodestyle --max-line-length=80 --statistics --ignore=E203,W503 eadf/*.py test/*.py
python3 setup.py test
Traceback (most recent call last):
  File "setup.py", line 21, in <module>
    from eadf import __version__
  File "/home/seb/git/Code/EADF/eadf/__init__.py", line 132, in <module>
    from .arrays import generateURA, generateULA, generateUCA, generateStackedUCA
  File "/home/seb/git/Code/EADF/eadf/arrays.py", line 33, in <module>
    from .eadf import EADF
  File "/home/seb/git/Code/EADF/eadf/eadf.py", line 56, in <module>
    from .core import evaluatePattern
  File "/home/seb/git/Code/EADF/eadf/core.py", line 26, in <module>
    import pint
  File "/home/seb/.local/lib/python3.8/site-packages/pint/__init__.py", line 28, in <module>
    from .measurement import Measurement
  File "/home/seb/.local/lib/python3.8/site-packages/pint/measurement.py", line 12, in <module>
    from .quantity import Quantity
  File "/home/seb/.local/lib/python3.8/site-packages/pint/quantity.py", line 21, in <module>
    from pkg_resources.extern.packaging import version
ModuleNotFoundError: No module named 'pkg_resources.extern'
make: *** [Makefile:12: test] Error 1
```

I try to come up with a solution.

EDIT:
seems like https://stackoverflow.com/a/19514783/3334355 has the solution.

SebastianSemper committed:
Fix minor bug in eval
